spec/element_locator_spec.rb

```diff
@@ -459,6 +457,68 @@
 
         expect(locate_all(selector)).to eq elements.last(2)
       end
+
+      context "and xpath" do
+        before do
+          expect(Watir).to receive(:prefer_css?).and_return(false)
+        end
+
+        it "converts a leading run of regexp literals to a contains() expression" do
+          elements = [
+            element(tag_name: "div", attributes: { class: "foo" }),
+            element(tag_name: "div", attributes: { class: "foob" })
+          ]
+
+          expect_all(:xpath, "(.//div)[contains(@class, 'fo')]").and_return(elements.last(1))
```

Comment:
Looks like I messed up the implementation for that example (should be `and_return(elements)` or include a non-matching element) but, yes, it should still iterate over each matching element and do the regexp comparison.